feat(react-query-dev-tools): Add ReactQueryDevToolsPanel support

toofff opened the pull request:
Hi :wave: 

Since major version 5 of TanstackQuery, the debug bar has been redesigned and much improved. However, support for the DevtoolsPanel component has been lost. The aim of my contribution is to reinstate this component.

![Tanstack-devtools-panel](https://github.com/user-attachments/assets/9ecc2be3-cb46-4dc8-b586-8cd867cf8d49)

### Work done in the **@tanstack/query-devtools** package:
- creation of a **contexts** folder to set up the 3 current contexts to lighten the `Devtools.tsx` file
- setting up `DevtoolsComponent` & `DevtoolsPanelComponent` private components
- setting up `TanstackQueryDevtools` & `TanstackQueryDevtoolsPanel` public components

### Work done in the **@tanstack/react-query-devtools** package:
- creating the `ReactQueryDevtoolsPanel` component
  - copy `ReactQueryDevtools` component to `ReactQueryDevtoolsPanel` 
  - removed `initialIsOpen` & `buttonPosition` props
  - add `isOpen` prop

### Others
- added documentation for the **ReactQueryDevtoolsPanel** component
- added `examples/react/devtools-panel` example

PS: Don't hesitate to hide white spaces to make proofreading easier.
![image](https://github.com/user-attachments/assets/474294e3-e4ba-4883-9fc5-eab5d700e1e9)

ardeora commented:
@toofff Thanks for working on the devtools panel API! Is this PR good to take a look at? Or do you need any help with it? 😄 I'll be happy to help you if you run into any blockers

toofff commented:
> @toofff Thanks for working on the devtools panel API! Is this PR good to take a look at? Or do you need any help with it? 😄 I'll be happy to help you if you run into any blockers

I haven't taken the time to finalize my PR yet, it's not good enough to look at.

The logic I had was to add 2 exportable components.
The Devtools component and the DevtoolsPanel component without the PiPProvider. Would this logic work for you?

I'm going to find some time this week to get back to it and hopefully finish it all off.

ardeora commented:
@toofff yeah that's fine. The devtools panel not having a PiP view makes sense because that would ideally be neet to be handled on the user side

toofff commented:
@ardeora I was able to work a bit last night and have a better idea where I'm headed.

The `@tanstack/react-query-devtools` package will export 2 components: `ReactQueryDevtools` & `ReactQueryDevtoolsPanel`.

The `@tanstack/query-devtools` package will also export 2 classes: `TanstackQueryDevtools` & `TanstackQueryDevtoolsPanel`.
These 2 classes will use the same `Devtools` component, but configured differently.

The `TanstackQueryDevtoolsPanel` class will configure the `Devtools` component so that:
- the open button is no longer present
- the close button is no longer present
- control buttons will probably no longer be present (e.g. position)
- the PIP button is no longer present

At first I just wanted to export the `DevtoolsPanel` component, but I ran into a lot more problems than I thought. The CSS was all broken, the functionalities more or less functional... So I think the simplest option is to simply disable certain functionality.

What do you think? Does the vision of my work suit you?

Note: If everything is OK for you, I could do the same thing for the other languages that the debug bar supports, but in another pull request :smile:

toofff commented on examples/react/simple/src/index.tsx:
This is just an example for me to work on, it will be removed soon and I will create a real example for this debug bar panel.

ardeora commented:
@toofff Thanks for working on this. This is looking great! I'll take a look at this today and detailed comments if required!

toofff commented:
@ardeora here's my finished contribution. Let me know what you think :pray:

ardeora commented:
Hi @toofff I took a look at the changes have made a PR to fix some things. https://github.com/toofff/tanstack-query/pull/1 
Let me know what you think! I think if everything looks good, we can merge this today/tomorrow whenever things go through

toofff commented:
@ardeora I'll review and test your pull request tomorrow evening, I don't think I'll have time tonight.

Thank you for your feedback.

toofff commented:
@ardeora to merge your PR I had to do a rebase and push on your repo but I didn't have the rights.
I simply retrieved your PR locally, squshed your 3 commits into 1 and cherry-picked that commit to put me in my branch. I sorted out all the conflicts and pushed.

I also corrected:
- the documentation to remove the isOpen prop from the `ReactQueryDevtoolsPanel` component
- updated the dependencies of the `@tanstack/query-example-react-devtools-panel` package with the latest versions of **Tanstack/Query**.
- kept the two **ReactQueryDevtools** & **ReactQueryDevtoolsPanel** files in the `@tanstack/react-query-devtools` package

toofff commented:
@ardeora if everything is good for you, you can merge ;)

Thank you very much for your help and cooperation in this adventure. :pray:

ardeora commented on packages/query-devtools/src/contexts/QueryDevtoolsContext.ts:
```diff
@@ -27,8 +26,10 @@ export interface QueryDevtoolsProps {
   buttonPosition?: DevtoolsButtonPosition
   position?: DevtoolsPosition
   initialIsOpen?: boolean
+  isOpen?: boolean
```
@toofff is this prop still required?

toofff replied:
Sorry, I forgot to delete it. Now I have.

ardeora reviewed (changes requested):
@toofff Everything else looks great! If you could just run `pnpm prettier:write` on the root of the project. The format CI will be happy!

toofff commented:
@ardeora everything is correct now, tonight I'll be able to start my new pull request for the other technologies and update the respective documentations and examples ;)

Thanks a lot for the help, it was much appreciated.

ardeora reviewed (approved):
@toofff Thank you so much for working on this. Apologies for taking so long to get it to merge but I appreciate it! Are you on Twitter/X? I would love to mention this feature and give you a shout-out.

toofff commented:
@ardeora yes, it is @yanoucrea
